porcupine/utils.py

```diff
@@ -645,12 +671,11 @@ def callback(shifted: bool, event: tkinter.Event[tkinter.Misc]) -> Any:
 
 # TODO: document this?
 def ask_encoding(text: str, old_encoding: str) -> str | None:
-    if porcupine.get_main_window().tk.call("winfo", "exists", ".choose_encoding"):
-        porcupine.get_main_window().tk.call("destroy", ".choose_encoding")
+    dialog = make_dialog_if_not_exists("choose_encoding")
+    if dialog is None:
+        # Another (possibly unrelated) encoding chooser is already running
+        return None
```

Comment:
```suggestion
    dialog = make_dialog_destroying_if_exists("choose_encoding")
    if dialog is None:
        # Another (possibly unrelated) encoding chooser is already running
        return None
```


Maybe we should remove this check, so if I open "choose encoding", then switch to another tab, and click the encoding button again, it will open a new dialog for that tab, and I don't have to close the existing window, and reopen it.